### ISSUE TITLE
Support Swift Package Manager (SPM) for iOS

### ISSUE THREAD

**Issue by tcwalther** [pytorch/pytorch#81552]
### 🚀 The feature, motivation and pitch

Currently, Pytorch needs to be added to an iOS project via Cocoapods. However, many newer projects have migrated to Swift Package Manager. It would be fantastic if one could add Pytorch mobile via SPM, too.

### Alternatives

Alternatively, I'd also be happy to vendor pytorch into my codebase. We threw out Cocoapods a while ago and don't really want to reintroduce it if possible.

**Comment by saugs** [comment]
Hi, this would be really nice! As I'm struggling adding PyTorch to a SwiftPackage library...

A possible easier solution would be to publish a compiled `xcframework` of the library, so that we can easily import it in SPM (cf. [here](https://stackoverflow.com/questions/63607450/how-can-i-add-a-local-library-as-a-dependency-in-swift-package-manager/63612618#63612618))

**Comment by dreiss** [comment]
If someone produces a PR that adds the necessary support, we could consider incorporating this into our release process.

**Comment by burak-initialcode** [comment]
https://blog.cocoapods.org/ 

[CocoaPods Trunk Read-only Plan](https://blog.cocoapods.org/CocoaPods-Specs-Repo)
TLDR: In two years we plan to turn CocoaPods trunk to be read-only. At that point, no new versions or pods will be added to trunk.

Last month I wrote about how CocoaPods is currently being maintained, I also noted that we were discussing converting the main CocoaPods spec repo "trunk" to be read-only:

We are discussing that on a very long, multi-year, basis we can drastically simplify the security of CocoaPods trunk by converting the Specs Repo to be read-only. Infrastructure like the Specs repo and the CDN would still operate as long as GitHub and jsDelivr continue to exist, which is pretty likely to be a very long time. This will keep all existing builds working.

I plan to implement the read-only mode so that when someone submits a new Podspec to CocoaPods, it will always be denied at the server level. I would then convert the "CocoaPods/Specs" repo to be marked as "Archived" on GitHub which should cover all of our bases.

Making the switch will not break builds for people using CocoaPods in 2026 onwards, but at that point, you're not getting any more updates to dependencies which come though CocoaPods trunk. This shouldn't affect people who use CocoaPods with their own specs repos, or have all of their dependencies vendored (e.g. they all come from npm.)
